AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m

```diff
@@ -141,12 +141,18 @@ - (MSDevice *)updatedDevice {
 #if TARGET_OS_IOS
     CTTelephonyNetworkInfo *telephonyNetworkInfo = [CTTelephonyNetworkInfo new];
     CTCarrier *carrier;
-    if (@available(iOS 12, *)) {
+    if (@available(iOS 12.1, *)) {
       NSDictionary<NSString *, CTCarrier *> *carriers = [telephonyNetworkInfo serviceSubscriberCellularProviders];
       for (NSString *key in carriers) {
         carrier = carriers[key];
         break;
       }
+    } else if (@available(iOS 12, *)) {
+        NSDictionary<NSString *, CTCarrier *> *carriers = [telephonyNetworkInfo valueForKey:@"serviceSubscriberCellularProvider"];
+        for (NSString *key in carriers) {
+            carrier = carriers[key];
+            break;
+        }
     }
```

Comment:
Code is shared.
```
for (NSString *key in carriers) {
        carrier = carriers[key];
        break;
      }
```
Can we maybe make it reusable?

Reply:
I was about to say the same thing

Reply:
Fixed